src/txmempool.h

```diff
@@ -63,8 +77,16 @@ struct LockPoints
 
 class CTxMemPoolEntry
 {
+public:
+    typedef std::reference_wrapper<const CTxMemPoolEntry> CTxMemPoolEntryRef;
+    // two aliases, should the types ever diverge
+    typedef std::set<CTxMemPoolEntryRef, CompareIteratorByHashGeneric> Parents;
+    typedef std::set<CTxMemPoolEntryRef, CompareIteratorByHashGeneric> Children;
```

Comment:
nit: By YAGNI, I would weigh against preemptive definitions

Reply:
I was doing some experiments where they diverge :)

Parents are fundamentally fixed by the TX (e.g., named by TXID) so there are some opportunities for eliminating the std::set based on that, whereas children kind of have to remain a std::set, but can potentially be turned into a pointer into MapNextTx.